src/clients/QBittorrent.ts

```diff
@@ -244,39 +240,53 @@ export default class QBittorrent implements TorrentClient {
 	 */
 	async generateCorrectSavePath(
 		searchee: Searchee,
-		{ save_path },
-	): Promise<{ save_path: string }> {
-		const subfolderContentLayout =
-			await this.isSubfolderContentLayout(searchee);
+		torrentInfo: TorrentInfo,
+	): Promise<string> {
+		const subfolderContentLayout = await this.isSubfolderContentLayout(
+			searchee,
+			torrentInfo,
+		);
 		if (subfolderContentLayout) {
-			return {
-				save_path: join(save_path, stripExtension(searchee.name)),
-			};
+			return dirname(torrentInfo.content_path);
 		}
-		return { save_path };
+		return torrentInfo.save_path;
+	}
+
+	/*
+	@return array of query results
+	 */
+	async getAllTorrentInfo(): Promise<TorrentInfo[]> {
+		const responseText = await this.request("/torrents/info", "");
+		return JSON.parse(responseText);
+	}
+
+	/*
+	@param hash the hash of the torrent or undefined for all torrents
+	@return array of query results
+	 */
+	async getTorrentInfo(hash: string): Promise<TorrentInfo[]> {
+		const torrents = await this.getAllTorrentInfo();
+		return torrents.filter(
+			(torrent) =>
+				hash === torrent.infohash_v1 || hash === torrent.infohash_v2,
```

Comment:
Is this just to future proof or what? I'm not entirely sure we have anything for v2 that would necessitate this?

Reply:
No it's needed right now. If a torrent has a v2 hash, it will truncate it to the first 40 characters (to match v1) and use that as the "hash" that gets checked for api calls and in BT_backup. But all torrents have a "hash" (always 40 characters), "infohash_v1" (naturally 40 characters) and "infohash_v2" (this is the full v2 hash, greater than 40 characters)".

I don't know how to allow the other endpoints to correctly get v2 torrents since they usually require a hash. Would probably get fixed by #471